apps/server/src/resolvers/updateReports.ts

```diff
@@ -0,0 +1,143 @@
+import { ReportInput } from '@/__generated__/resolvers-types'
+import { modelTypes } from '@/fistbump-types'
+import Cycle from '../lib/mongoose/models/Cycle'
+import Report from '..//lib/mongoose/models/Report'
+import { ApolloContext } from '@/types'
+
+export async function resolveUpdateReport(
+  _: any,
+  {
+    targetId,
+    cycleId,
+    input,
+  }: {
+    targetId: String
+    cycleId: String
+    input: ReportInput
+  },
+  context: ApolloContext
+) {
+  if (!cycleId) {
+    const cycle = await Cycle.getCurrentCycle()
+    if (cycle === null) throw new Error('No current cycle found')
+    cycleId = cycle._id.toString()
+  }
+
+  const report = await Report.findOne({
+    '_id.targetId': targetId,
+    '_id.cycleId': cycleId,
+  })
+
+  if (!report) throw new Error('No report found')
+  if (!context.user?.id || !context.user.email) throw new Error('Unauthorized')
+
+  const isSelf = context.user?.id === targetId
+  const isManager =
+    context.user?.id === report.reviews.manager.reviewerId?.toString()
+  const isReviewer = report.reviews.peers.some(
+    (peer) => peer.reviewerId?.toString() === context.user?.id
+  )
+
+  if (!isSelf && !isManager && !isReviewer) throw new Error('Unauthorized')
+
+  // //if myself -> i can mutate self review or nominate peers!
+  if (isSelf) {
+    if (report.status === modelTypes.REPORT_STATUS.NOMINATION) {
+      if (input.reviews?.peers?.reviewerId) {
+        const peersToNominate = report.reviews.peers.find(
+          (peer) => peer.reviewerId === null
+        )
+        if (peersToNominate) {
+          if (input.reviews.peers.reviewerId)
+            peersToNominate.reviewerId = new modelTypes.ObjectId(
+              input.reviews.peers.reviewerId
+            )
+        }
+        await report.save()
+
+        const peerIds = report.reviews.peers.map((peer) => {
+          return {
+            reviewerId: peer.reviewerId,
+          }
+        })
+
+        return {
+          reviews: {
+            peers: peerIds,
```

Comment:
This is a bit confusing. I would expect `peerIds` to be an array of strings. But it's actually an array of reviews that only contain `reviewerId`. 

This kind of stuff isn't urgent to fix... as long as someone-besides-you doesn't have to make any changes to this later.

Reply:
discussed